core/src/Kokkos_TBB.hpp

```diff
@@ -0,0 +1,1866 @@
+      }
+
+      return *this;
+    }
+
+    pointer_type pointer() const {
+      return reinterpret_cast<pointer_type>(m_value_buffer);
+    }
+
+    reference_type reference() const {
+      return ValueOps::reference(
+          reinterpret_cast<pointer_type>(m_value_buffer));
+    }
+  };
+
+public:
+  void execute() const {
+    std::size_t value_size =
+        Analysis::value_size(ReducerConditional::select(m_functor, m_reducer));
+
+    value_type_wrapper final_value(value_size);
+    value_type_wrapper identity(value_size);
+
+    ValueInit::init(ReducerConditional::select(m_functor, m_reducer),
+                    final_value.pointer());
+    ValueInit::init(ReducerConditional::select(m_functor, m_reducer),
+                    identity.pointer());
+    
+    using RangeType = tbb::blocked_range<decltype(m_policy.begin())>;
+    tbb::this_task_arena::isolate([this, &final_value,&identity]{
+        final_value = tbb::parallel_reduce(RangeType(m_policy.begin(),
+                                                     m_policy.end(),
+                                                     m_policy.chunk_size()),
+                                           identity,
+                                           [this](const RangeType& r, value_type_wrapper const &init) {
+                                             auto update = init;
+                                             execute_functor_range<WorkTag>(update.reference(), r.begin(), r.end());
+                                             return update;
+                                           },
+                                           [this](value_type_wrapper a,
+                                                  value_type_wrapper const &b)  {
+                                             ValueJoin::join(
+                                                             ReducerConditional::select(m_functor, m_reducer),
+                                                             a.pointer(), b.pointer());
+                                             return a;
+                                           }
+                                           );
+      }); //isolate
+
+    pointer_type final_value_ptr = final_value.pointer();
+
+    Kokkos::Impl::FunctorFinal<ReducerTypeFwd, WorkTagFwd>::final(
+        ReducerConditional::select(m_functor, m_reducer), final_value_ptr);
+
+    if (m_result_ptr != nullptr) {
+      const int n = Analysis::value_count(
+          ReducerConditional::select(m_functor, m_reducer));
+
+      for (int j = 0; j < n; ++j) {
+        m_result_ptr[j] = final_value_ptr[j];
+      }
+    }
+    
+  }
+
+  template <class ViewType>
+  inline ParallelReduce(
+      const FunctorType &arg_functor, Policy arg_policy,
+      const ViewType &arg_view,
+      typename std::enable_if<Kokkos::is_view<ViewType>::value &&
+                                  !Kokkos::is_reducer_type<ReducerType>::value,
+                              void *>::type = NULL)
```

Comment:
`nullptr`? Or is `NULL` their preference?

Reply:
It looks like all the other backends use `NULL` for the equivalent line in that code.

Reply:
Ok